xchange-stream-ftx/src/main/java/info/bitrich/xchangestream/ftx/FtxStreamingAdapters.java

```diff
@@ -0,0 +1,94 @@
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.marketdata.OrderBook;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.instrument.Instrument;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.zip.CRC32;
+
+public class FtxStreamingAdapters {
+
+    private final static ObjectMapper mapper = StreamingObjectMapperHelper.getObjectMapper();
+
+    public static OrderBook adaptOrderbookMessage(OrderBook orderBook, Instrument instrument, JsonNode jsonNode){
+
+        Streams.stream(jsonNode)
+                .filter(JsonNode::isObject)
+                .map(res->{
+                    try{
+                        return mapper.readValue(res.toString(),FtxOrderbookResponse.class);
+                    }catch (IOException e){
+                        throw new RuntimeException(e);
+                    }
+                })
+                .forEach(
+                        message -> {
+
+                            if(message.getAction().equals("partial")){
+                                message.getAsks().forEach(ask->{
+                                    orderBook.getAsks().add(new LimitOrder.Builder(Order.OrderType.ASK,instrument).limitPrice(ask.get(0)).originalAmount(ask.get(1)).build());
+                                });
+
+                                message.getBids().forEach(bid ->{
+                                    orderBook.getBids().add(new LimitOrder.Builder(Order.OrderType.BID,instrument).limitPrice(bid.get(0)).originalAmount(bid.get(1)).build());
+                                });
+                            }else{
+                                message.getAsks().forEach(ask->
+                                        orderBook.update(new LimitOrder.Builder(Order.OrderType.ASK,instrument).limitPrice(ask.get(0)).originalAmount(ask.get(1)).build())
+                                );
+                                message.getBids().forEach(bid->
+                                        orderBook.update(new LimitOrder.Builder(Order.OrderType.BID,instrument).limitPrice(bid.get(0)).originalAmount(bid.get(1)).build())
+                                );
+                            }
+
+                            if(orderBook.getAsks().size() > 0 && orderBook.getBids().size() > 0){
+                                Long calculatedChecksum = getOrderbookChecksum(orderBook.getAsks(),orderBook.getBids());
+
+                                if(!calculatedChecksum.equals(message.getChecksum())){
+                                    throw new RuntimeException("Checksum is not correct!");
+                                }
+                            }
+                        });
+
+        return new OrderBook(Date.from(Instant.now()), Lists.newArrayList(orderBook.getAsks()),Lists.newArrayList(orderBook.getBids()),true);
```

Comment:
new ArrayList<>(orderbook.getasks()) no need for extra google dep

Reply:
done